### ISSUE TITLE
Allow selection of Steam Linux Runtime in Game List

### ISSUE THREAD

**Issue by xperia64** [DavidoTek/ProtonUp-Qt#328]
**Is your feature request related to a problem? Please describe.**
Currently, Steam hides the `Soldier (2.0)` and `Sniper (3.0)` Steam Linux Runtime compatibility tools in the Compatibility tab of games and does not allow them to be selected manually; only `Scout (1.0)` can be chosen.

While the primary purpose of ProtonUp-Qt is managing versions of Proton, it already has the necessary infrastructure in place to configure arbitrary compatibility tools for games.

**Describe the solution you'd like**
I would like the Compatibility Tool dropdown menus to show the `steamlinuxruntime` variants, or at least have the option for them to appear.

**Describe alternatives you've considered**
Editing `config/config.vdf` manually via a text editor works, but is inconvenient.

**Additional context**
I am opening this as an issue rather than a PR as I'm not sure if you want to always show the Steam Linux Runtimes or have some checkbox somewhere.

Additionally, while `get_steam_ctool_list` in `steamutil.py` already has an optional argument to allow showing non-Proton compatibility tools, Steam Linux Runtimes are not currently enumerated as compatibility tools even if that is set to True.

The section of `update_steamapp_info` that handles Steam Linux Runtimes could be updated as follows to address this:
```py
elif 'Steam Linux Runtime' in a.game_name:
    a.app_type = 'runtime'
    if steam_app.get('appid') in ctool_map:
        ct = ctool_map.get(steam_app.get('appid'))
        a.ctool_name = ct.get('name')
        a.ctool_from_oslist = ct.get('from_oslist')
```

**Comment by sonic2kk** [comment]
Steam Linux Runtime 1.0 is for Native Linux games only, 2.0 and 3.0 are for Proton games only. Steam does not show these because they cannot be used as standalone compatibility tools. Only 1.0 Scout can be used for games and this is for native games only.

ProtonUp-Qt should already be showing the Steam Linux Runtime 1.0 on the Games List. I was just able to use it to select the Steam Linux Runtime for Half-Life:

![image](https://github.com/DavidoTek/ProtonUp-Qt/assets/7917345/99357372-c16e-4081-98ea-541e5d4847d9)

It does now show 2.0 and 3.0 because, as mentioned, these cannot be used for games regularly.

The launch command for 2.0 and 3.0 are prepended to a Proton game's launch command. Proton flavours based on Proton 5.13 and above enforce the use of the Steam Linux Runtime, and in their files they have a text file called `toolmanifest.vdf`. This has a section called `require_tool_appid` which denotes the AppID of the Steam Linux Runtime required for that tool.

Steam uses this for two things:
1. To download the relevant tool (i.e. the relevant SLR if it is not downloaded) before a game launch, so if a tool requires SLR Sniper 3.0, it will make sure that is downloaded before a game is launched
2. To know how to find the relevant compatibility tool path to prepend to the game launch command

For example, GE-Proton8-25 uses SLR 3.0 Sniper (and Valve Proton 8, in fact, I think all Proton 8 variants use Sniper). Here is what the start command for a game using SLR 3.0 Sniper looks like:

```
/run/media/emma/500GB SSD/Games/steamapps/common/SteamLinuxRuntime_sniper/_v2-entry-point --verb=waitforexitandrun -- /run/media/emma/500GB SSD/Games/steamapps/common/Proton 8.0/proton waitforexitandrun /run/media/emma/500GB SSD/Games/steamapps/common/CHAOS;CHILD/boot.bat
```

SLR 1.0 Scout is appended in a similar way, but they are used differently. Only Scout 1.0 can be used as a compatibility tool, the 2.0 and 3.0 containers are only for use as part of a launch command (i.e. to launch games inside the given pressure-vessel container). Their file structure and purpose are different, you can inspect this yourself with "Browse local files" from the Steam Client (you can search "Steam Linux Runtime" or the specific AppIDs in the Steam library search).

<hr>

In short, Scout 1.0 is already supported, and 2.0 and 3.0 cannot be supported because they are not compatibility tools.

**Comment by sonic2kk** [comment]
It is indeed possible to run games in 2.0 and 3.0 but it is not intended usage, otherwise Steam would allow you to do it :-)

**Comment by xperia64** [comment]
While I acknowledge that 2.0 is discouraged as a native Linux target, 3.0 is intended to be the successor for 1.0, and there are a handful of released titles already targeting it: https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/-/blob/main/docs/slr-for-game-developers.md#native-linux-games-targeting-steam-runtime-3-sniper

I confirmed RetroArch actively uses Sniper, and Endless Sky is listed as Playable.

**Comment by sonic2kk** [comment]
When the Steam Client starts using Sniper instead of Scou, or displays both, then ProtonUp-Qt could probably pick it up. But while the Steam Client doesn't display it I'm not sure ProtonUp-Qt should either... That's just my opinion though.
